packages/bitcoin-dlc-provider/lib/BitcoinDlcProvider.ts

```diff
@@ -1052,14 +1219,21 @@ Payout Group not found',
       .reduce((acc, val, i) => acc + Number(val) * base ** i, 0);
 
     const piecesSorted = payoutFunction.pieces.sort(
-      (a, b) => Number(b.endpoint) - Number(a.endpoint),
+      (a, b) => Number(a.endpoint) - Number(b.endpoint),
```

Comment:
What's the rationale for changing this?

Reply:
I think this is a good catch!

The line below relies on the array being in ascending endpoint order, to find the right piece for the outcome.
`const piece = piecesSorted.find((piece) => outcome < piece.endpoint);`

In our previous code, it would've sorted the pieces in reverse orde. (I think this logic was carried over from finding the proper rounding interval, which does in fact require it to be in reverse)


**Example:** You have three pieces of (start, endpoint): 
-  one for (0,30), one for (30, 60) and one for (60, 90)

The `piecesSorted.find` finds the first element in the array that that has an endpoint greater than outcome.

If it was reversed `[(60, 90), (30, 60), (0,30))`, the payout curve piece for outcome `25` would've been `(60,90)`.
The correct answer is (0,30)

---

nit:
Right now, we're doing a search for a piece using a linear scan which works. Ideally, we do a binary search and follow spec:

https://github.com/discreetlogcontracts/dlcspecs/blob/2095ba2a6455ba8493555b958c8dcdb49313a36c/PayoutCurve.md?plain=1#L119-L123

Just leaving reference for the future, since we won't be using a large amount of pieces any time soon.

Reply:
Yup exactly as Steven described ^. We were finding the wrong piece earlier since we were looking in reverse order. Swapping the order fixes that issue!